cmd/dockerd/daemon.go

```diff
@@ -301,10 +302,15 @@ func (cli *DaemonCli) start() (err error) {
 
 func (cli *DaemonCli) reloadConfig() {
 	reload := func(config *daemon.Config) {
+
+		// Reload authorization plugin
+		cli.authzMiddleware.SetPlugins(config.AuthorizationPlugins)
```

Comment:
the configuration locks concurrent reloads, there cannot be two `SetPlugins` called at the same time. However, it can be racy with the reading operation. I don't think that's as problematic as two concurrent writes.